src/reducers/cities.reducer.js

```diff
@@ -0,0 +1,10 @@
+import { citiesConstants} from '../constants';
+
+export function cities(state=[],action){
+    switch (action.type){
+        case citiesConstants.GET_ALL:
+            return action.cities.length?action.cities:[];
```

Comment:
if `cities.length === 0` then `action.cities === []` so you can rewrite whole line as `return action.cities`